coral-schema/src/test/resources/base-complex-nullable-with-defaults.avsc

```diff
@@ -7,7 +7,7 @@
     "type" : "int"
   }, {
     "name" : "Int_Field",
-    "type" : [ "null", "int" ],
+    "type" : [ "int", "null" ],
```

Comment:
can you add a UT where we union `basecomplexnullablewithdefaults` with itself and some of the underlying fields have incorrect ordering for nullable union types? (basically, retaining the origin basecomplexnullablewithdefaults and union with self)?

(1) I see that during the merge of LHS and RHS schemas, if the datatype ordering in the nullable union is not correct, we fix it. it'll be good to have a UT validating it. 
(2) this seems like a design choice where if we are union-ing 2 incorrect base tables, we fix the datatype ordering of the merged schema on the fly (by calling `reorderOptionIfRequired`), but we don't fix the schema if we are generating schema of a single incorrect base table.  Should we be consistent in our approach?

Reply:
(1) We can't create a base table with incorrect ordering for nullable union types in UT given Avro 1.10 doesn't allow it
(2) We call `reorderOptionIfRequired` for `unionFieldSchema` because it is generated by our code, we need to ensure its correctness. But for a single base table, it should be user's responsibility to ensure the correctness of its schema.